### DISCUSSION TITLE
RFC 2024-02-11 Channels

### DISCUSSION THREAD

**Discussion by code8buster** [meshtastic/rfcs#3] [General]
Discuss this RFC in this thread if you don't want it to get lost on Discord/Discourse/etc!
https://github.com/meshtastic/rfcs/blob/main/rfcs/2024-02-11-channels.md

With respect to the use of the word "channel" to denote the separately encrypted broadcasts, termed in the proposal "Meshtastic Communications Group", I believe this function is extremely similar to "talkgroups" on DMR. It creates virtual networks on top of a shared medium, the radio's frequency channel.

"A DMR talkgroup is simply a way of grouping many Radio IDs into a single digital contact. Or put another way, a talkgroup is a method of organizing radio traffic specific to the DMR users that all want to hear the same thing and not be bothered by other radio traffic on a DMR network that they are not interested in hearing." from https://www.dmrfordummies.com/talkgroups/

Hopefully we'll work [this 3.0 project item](https://github.com/orgs/meshtastic/projects/20#) out here

**Comment by rcarteraz** [comment]
**Times in MST**

MC Hamster — Today at 4:35 PM
Started a document with proposals for our overuse of the word 'channel'.
[GitHub - rfcs/rfcs/2024-02-11-channels.md at main · meshtastic/rfcs](https://github.com/meshtastic/rfcs/blob/main/rfcs/2024-02-11-channels.md)
@garth fyi
Feedback very welcome.
The operating principle I started with was the word 'channel' is used by the ham community to define a frequency allocation within a band plan. Rather than reinventing the wheel, we can stick to that.
Everything else has been moved away from using 'channel' to something else.
We also hear about people "joining a Meshtastic network". Let's use this as part of our lexicon and codify it.

nagu — Today at 4:41 PM
Network, mesh, chatroom
I'd use the term "mesh"
It fits

MC Hamster — Today at 4:42 PM
The term mesh for which use case?

nagu — Today at 4:43 PM
Maybe not mesh. That's already in use. It's 2am here.
To replace 'channel' in the non-frequency band context.
You're right. Group. That's good.

Jonathan Bennett — Today at 4:43 PM
"Mesh" is the group of devices using the same LoRa config.

nagu — Today at 4:44 PM
Yeah, it's late and I wasn't thinking.

codebuster — Today at 4:44 PM
Was the intention to create the discussion on the discussion tab of the rfcs repo?

Jonathan Bennett — Today at 4:44 PM
I sort of like "room" to describe a shared password config.

codebuster — Today at 4:44 PM
I'd like to put something where it won't get lost in the Discord churn so I figure that's probably the best place.

MC Hamster — Today at 4:45 PM
Go for it 🙂

nagu — Today at 4:45 PM
Room feels very 90s.

MC Hamster — Today at 4:45 PM
My intention was to bring visibility to the topic.
🙂

Jonathan Bennett — Today at 4:45 PM
On brand then? :p

nagu — Today at 4:45 PM
I think group is good. Similar to a WhatsApp group or a Signal group.

Jonathan Bennett — Today at 4:45 PM
That works too.

MC Hamster — Today at 4:45 PM
I think discussion here is good as long as it's eventually documented somewhere.

nagu — Today at 4:45 PM
People already know what group means.

MC Hamster — Today at 4:46 PM
Hehe... This is the "C" process of the RFC.

nagu — Today at 4:46 PM
I don't know that I like "Meshtastic Network" to mean a QR code / URL.
That's just confusing, IMO.
Call it the Group Link.

MC Hamster — Today at 4:47 PM
Well... I think the first thing to do is agree that the use cases and description are correct. I think that's captured.
Is it accurate?

nagu — Today at 4:48 PM
I'd say so.

MC Hamster — Today at 4:48 PM
Cool. As long as the use cases and description are right, then we can work out the names. 🙂 🙂 🙂

codebuster — Today at 4:48 PM
As proposed, it's wordy.
We could steal from the digital radio (amateur digital radio) world.
Meshtastic Comm. Group == talkgroups, in DMR world.

nagu — Today at 4:49 PM
I think the important thing is that it be names that the average user instinctually gets.

codebuster — Today at 4:49 PM
"A DMR talkgroup is simply a way of grouping many Radio IDs into a single digital contact. Or put another way, a talkgroup is a method of organizing radio traffic specific to the DMR users that all want to hear the same thing and not be bothered by other radio traffic on a DMR network that they are not interested in hearing."

MC Hamster — Today at 4:49 PM
Propose otherwise. This is just what I threw together in 10 minutes.

nagu — Today at 4:49 PM
Like group is used in other chat app contexts for a similar use case.

codebuster — Today at 4:49 PM
I know 😉 I'm writing the post now.

MC Hamster — Today at 4:49 PM
Rad.

nagu — Today at 4:51 PM
My feelings on new names:

RF Spectrum Channel: Channel
Meshtastic communication channel: Group or Group Chat
Application Channel: Group Link
I think they're all self-explanatory. Alternately, RF channel could be "Band" and the comms channel could be "Channel," but I think that's less similar to current standard terminology and will be more confusing for users. Talkgroup is pretty great too.
codebuster — Today at 4:53 PM
https://github.com/meshtastic/rfcs/discussions/3

MC Hamster — Today at 4:58 PM
Good point on talkgroup from DMR.

nagu — Today at 4:59 PM
It also, frankly, sounds good.

pdxlocs — Today at 4:59 PM
I don't mind calling them 'chat channels' and 'frequency channels'. Yes, it can be confusing at first. There are many aspects of Meshtastic that are more confusing.

nagu — Today at 4:59 PM
makes it less accessible

MC Hamster — Today at 5:00 PM
I think the only use of channel is to identify which frequency within a band plan.

nagu — Today at 5:00 PM
is or should be

pdxlocs — Today at 5:00 PM
This is a standard definition that is widely used.

MC Hamster — Today at 5:00 PM
should be.

codebuster — Today at 5:01 PM
It's harder when you're talking to people who don't have a technical clue.

MC Hamster — Today at 5:01 PM
or must be

nagu — Today at 5:01 PM
At the end of the day, the goal (tell me if I'm wrong) is more widespread adoption. That means more user-friendliness, as Meshtastic becomes more mature and part of that is terminology, imo.

pdxlocs — Today at 5:02 PM
These people have no business setting the frequency channel.

codebuster — Today at 5:02 PM
at least I've had a hard time connecting people with the ideas that each use of the same word represents

MC Hamster — Today at 5:02 PM
Agreed. This is something we should abstract away from the conversation.

nagu — Today at 5:03 PM
True! But if they're told "you're in the wrong channel," they shouldn't have to guess at which of three definitions you mean, and while doing so, mess up their config.

codebuster — Today at 5:03 PM
anyhow... I'm willing to comb through all the firmware to change the terminology ;P

Jonathan Bennett — Today at 5:03 PM
Having overlapping terminology is one of the ways you get noobs changing their frequency channel settings. ><

pdxlocs — Today at 5:03 PM
They will mess it up either way. Sorry, I'm just arguing for arguing's sake. I'll stop.

nagu — Today at 5:04 PM
KISS. Keep It Simple, Sammy.

Jonathan Bennett — Today at 5:04 PM
And yes, @pdxlocs is right, noobs gonna noob.

nagu — Today at 5:04 PM
lol

Jonathan Bennett — Today at 5:04 PM
But maybe hopefully a little bit less this way.

pdxlocs — Today at 5:05 PM
This may neither be here nor there, but I can imagine an app. Meshtastic Lite, that ONLY has chat capability. No settings.

nagu — Today at 5:05 PM
YES. This is a long-term idea, but yes!

codebuster — Today at 5:05 PM
like you just scan the QR and it presents nothing but the chats or you hand someone a preconfig'd device

pdxlocs — Today at 5:06 PM
Yes, just comment out the part of the code that goes to the settings.

Jonathan Bennett — Today at 5:06 PM
I could see that being a thing.

pdxlocs — Today at 5:06 PM
Again, I'm off-topic, stepping back.

Jonathan Bennett — Today at 5:07 PM
That's sort of what managed mode is for.

codebuster — Today at 5:07 PM
it's an interesting idea. make note of it wherever you want to keep notes.

nagu — Today at 5:07 PM
Honestly, looking at settings - I don't see a single setting that a lite user would need. I keep thinking that maybe there is one... but the lite version would be JUST chat. That's it. Maybe the ability to create a new talkgroup (well done @codebuster!) excellent idea.

codebuster — Today at 5:09 PM
#notmyidea :}

nagu — Today at 5:09 PM
no I mean talkgroup was yours

codebuster — Today at 5:12 PM
eh, others brought it up before. I just happened to be the one to remember it at the right time. thanks though!

nagu — Today at 5:14 PM
alright, goodnight. it's laaaate night here.

Tommy — Today at 5:16 PM
Definitely feels like this conversation should be on the RFC itself. it's fine to chat about it here, but I would assume that unless it's written on the RFC it will not be considered "official".

codebuster — Today at 5:18 PM
yah. people put anything in particular you want to memorialize in the discussion board about this rfc. https://github.com/meshtastic/rfcs/discussions/3

MC Hamster — Today at 5:20 PM
I don’t think the name of “Meshtastic light” sounds very good. What about “Meshtastic extra snug fit”. OK not going to derail the conversation.

nagu — Today at 5:20 PM
Honestly, just a lite mode for the app. When you first install, you choose a mode. Lite or Advanced. okay sleep bye.

MC Hamster — Today at 5:21 PM
Completely on board with this.

pdxlocs — Today at 5:21 PM
Maybe just Simple or Advanced. Sorry I ever said Lite. Now off to learn Kotlin and Swift. brb.

Jonathan Bennett — Today at 5:23 PM
Or just hide the settings when connecting to a node in managed mode. Add a pin to disable managed?

pdxlocs — Today at 5:26 PM
The only thing I don't like about talkgroup is it makes me think audio, not text.

garth — Today at 5:51 PM
Channels are better than talk groups, only hams know talk group.

Crichton — Today at 5:59 PM
But are the app developers?

pdxlocs — Today at 6:06 PM
I personally have nothing against channels. I've said it before, most apps don't give you access to the front (input) and back (output) end. Meshtastic does. For better or worse. It's pretty obvious lots of users are confused, so personal feelings aside, it seems to make sense to come up with a new term for the chat channels.

Crichton — Today at 6:08 PM
I don’t mind changing it but it’s obviously a big lift and eco-system wide. If next breaking change is 3.0 and that’s 18 months off, is it still worth it?

garth — Today at 6:08 PM
Super not on board. Especially for a phone app. We need to cull 30% of our settings. Not hide them.

garth — Today at 6:12 PM
That is not the main Meshtastic use though. If we are going to fundamentally change the thing everyone associates with Meshtastic the name has to be better. Meshtastic Communications Channel is just three times as long, everyone will still call it channel.

pdxlocs — Today at 6:20 PM
AOL nailed it: Chat Room or bust. (bust).

**Comment by tekstrand** [comment]
Thank you @mc-hamster for starting this great discussion! I think it has already invited great input from the community including the devs, contributors, and regular users of Meshtastic.

After the conversation that has already happened, I believe the community would benefit from a modified approach to this conversation. Instead of starting with any solution, I suggest that we begin by clearly defining the problem. I believe that @rcarteraz is already thinking this way as well, based on his input in Discord after what he posted above.

In the interest of that proposal, I've taken a first pass at a draft which helps us reframe the conversation.

## RFC
This Request for Comments (RFC) seeks to address a recurring concern within the Meshtastic community regarding the use of the term "channel." Meshtastic employs "channel" to describe various concepts, including frequency allocations and group communication spaces. This multiplicity of meanings has led to confusion among users, especially those new to the platform.

### Problem Statement

The term "channel," while deeply ingrained in both amateur radio and Meshtastic vernacular, has been identified as a source of ambiguity. It is traditionally used to denote a specific frequency allocation within a band plan. However, Meshtastic's innovative use extends beyond this definition, encompassing communication spaces among users. This dual usage complicates user interactions, potentially hindering the adoption and effective use of the platform.

### Objective

The primary objective of this RFC is to **initiate a community-driven discussion** aimed at reevaluating the term "channel." By addressing this terminology concern, we aim to enhance clarity, reduce confusion, and improve the overall user experience on the Meshtastic network.

### Considerations

1. Terminological Clarity: Any new term should unambiguously describe its intended concept without overlap with existing radio or communication terms.
2. User Familiarity: The replacement should leverage terminology familiar to both new and existing users, facilitating ease of adoption.
3. Technical Accuracy: The term should accurately reflect the technical nature of Meshtastic's features and functionalities.
4. Community Feedback: Emphasizing a collaborative approach, this RFC encourages feedback and suggestions from the Meshtastic community.

### Call to Action

We invite all community members to contribute their perspectives, suggestions, and insights regarding the problematic use of "channel". This collaborative effort will ensure that any terminology changes will serve to enhance understanding, usability, and community engagement with Meshtastic.

This RFC represents an opportunity to refine our language and align more closely with our platform's innovative spirit. Your participation is crucial to achieving a consensus that reflects the diverse experiences and needs of the Meshtastic community.

**Comment by charminULTRA** [comment]
After reviewing the template, I agree that @tekstrand's format is preferable. Although terminology/language changes could affect the firmware/apps/docs/protos pervasively, it's critical that we take a user-based (UX) focused approach with changes like this. The current template seems to be too focused on technical implications for it to be appropriate for this RFC.
